velox/exec/tests/MarkDistinctTest.cpp

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/exec/tests/utils/OperatorTestBase.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+#include "velox/exec/tests/utils/QueryAssertions.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::exec::test;
+
+class MarkDistinctTest : public OperatorTestBase {
+ public:
+  template <class NativeType>
+  std::function<VectorPtr()> baseFunctorTemplate() {
+    return [&]() {
+      return makeFlatVector<NativeType>(
+          2, [&](vector_size_t row) { return (NativeType)(row % 2); }, nullptr);
+    };
+  }
+
+  template <class NativeType>
+  void markDistinctTest(
+      std::function<VectorPtr()> baseFunctor,
+      std::function<TypePtr()> exprTypeFunctor = []() {
```

Comment:
- exprTypeFunctor parameter is not used; remove
- replace baseFunctor with const VectorPtr&
- See example in "review comments" commit below:

https://github.com/facebookincubator/velox/compare/main...mbasmanova:velox-1:ossmarkdistinct?expand=1